repose-aggregator/core/repose-core/src/main/scala/org/openrepose/powerfilter/ReposeFilterLoader.scala

```diff
@@ -0,0 +1,321 @@
+    try {
+      logger.info("Getting child application context for {} using classloader {}", filterClassName, filterClassLoader.toString)
+      val filterContext = CoreSpringProvider.getContextForFilter(applicationContext, filterClassLoader, filterClassName, getUniqueContextName(filter))
+      //Get the specific class to load from the application context
+      val filterClass: Class[_] = filterClassLoader.loadClass(filterClassName)
+      val newFilterInstance: Filter = try {
+        filterContext.getBean(filterClass).asInstanceOf[Filter]
+      } catch {
+        case e: NoSuchBeanDefinitionException =>
+          logger.debug("Could not load the filter {} using Spring. Will try to manually load the class instead.", filterClassName, e)
+          //Spring didn't load the filter as a bean, try manually creating a new instance of the class
+          val rtnFilterInstance = filterClass.newInstance.asInstanceOf[Filter]
+          //Add the instance to the application context using its full class name
+          filterContext.getBeanFactory.registerSingleton(rtnFilterInstance.getClass.getName, rtnFilterInstance)
+          rtnFilterInstance
+      }
+      newFilterInstance.init(new FilterConfigWrapper(servletContext, filterType, filter.getConfiguration))
+      logger.info("Filter Instance: {} successfully created", newFilterInstance)
+      FilterContext(newFilterInstance, filter.getName, filter.getFilterCriterion.evaluate, filterContext, filter)
+    } catch {
+      case e: ClassNotFoundException =>
+        throw new FilterInitializationException(s"Requested filter, $filterClassName does not exist in any loaded artifacts", e)
+      case e: ServletException =>
+        val message = s"Failed to initialize filter $filterClassName"
+        logger.error(message)
+        throw new FilterInitializationException(message, e)
+      case e: ClassCastException =>
+        throw new FilterInitializationException(s"Requested filter, $filterClassName is not of type javax.servlet.Filter", e)
+      case e@(_: InstantiationException | _: IllegalAccessException) =>
+        throw new FilterInitializationException(s"Requested filter, $filterClassName is not an annotated Component nor does it have a public zero-argument constructor.", e)
+    }
+  }
+
+  private def getUniqueContextName(filterInfo: FilterConfig): String = {
+    val sb = new StringBuilder
+    if (filterInfo.getId != null) sb.append(filterInfo.getId).append("-")
+    sb.append(filterInfo.getName).append("-")
+    sb.append(UUID.randomUUID.toString)
+    sb.toString
+  }
+}
+
+object ReposeFilterLoader {
+
+  case class FilterContext(filter: Filter, filterName: String, shouldRun: HttpServletRequestWrapper => Boolean, appContext: AbstractApplicationContext, filterConfig: FilterConfig)
+
+  class FilterContextList(val registrar: FilterContextRegistrar, val filterContexts: List[FilterContext], val bypassUriRegex: Option[String]) extends AutoCloseable {
```

Comment:
Sure. This was the end result of a lot of refactoring and it could go one more step. @adrianjgeorge and I talked about this and it slipped past me again.
Updated.